storage/allocator_test.go

```diff
@@ -290,6 +290,52 @@ func TestAllocatorExistingReplica(t *testing.T) {
 	}
 }
 
+// TestAllocatorRelaxConstraints verifies that attribute constraints
+// will be relaxed in order to match nodes lacking required attributes,
+// if necessary to find an allocation target.
+func TestAllocatorRelaxConstraints(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	s, _, stopper := createTestStore(t)
+	defer stopper.Stop()
+	gossipStores(s.Gossip(), multiDCStores, t)
+
+	testCases := []struct {
+		required         []string // attribute strings
+		existing         []int    // existing store/node ID
+		relaxConstraints bool     // allow constraints to be relaxed?
+		expID            int      // expected store/node ID on allocate
+		expErr           bool
+	}{
+		{[]string{"a", "ssd"}, []int{}, true, 1, false},
```

Comment:
wow, this was really hard to figure out. A reference to `multiDCStores` would make this much easier to dereference